android/KMEA/app/src/main/java/com/keyman/engine/KeyboardPickerActivity.java

```diff
@@ -252,7 +253,11 @@ protected void onResume() {
       adapter.notifyDataSetChanged();
     }
 
-    int curKbPos = KeyboardController.getInstance().getKeyboardIndex(KMKeyboard.currentKeyboard());
+
+    SharedPreferences prefs = this.getSharedPreferences(this.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
+    String currentKeyboard = KMKeyboard.currentKeyboard();
+    int curKbPos = (currentKeyboard == null) ? prefs.getInt(KMManager.KMKey_UserKeyboardIndex, 0) :
```

Comment:
curKbPos is pretty unreadable and unpronounceable -- since we are not limited to 8-character variable names pick a name that describes what this really is and might pass spell check if it weren't camel case
I'm guessing it is the index into the keyboard list for the currently selected keyboard? Maybe -- but I had to think about it for a minute. Kb usually means kilobytes, and a cur is a mean dog.

Reply:
ok - I'll split this into if-else and rename `curKbPos` to `currentKeyboardIndex`

Reply:
looks good -- one other question: if the prefs don't exist and it returns the default of zero, will that crash? or is zero always ok because you always have at least one keyboard in the list?

Reply:
There will always be at least one keyboard in the list, so 0 points to the first keyboard in the list.

The KeyboardPicker feature to delete keyboards is only available when there's 2+ keyboards.